### PR TITLE
[BUILD] Migrate gradle from groovy -> kotlin

### PR DESCRIPTION
**Pull request by m273d15**
The Kotlin DSL is statically typed and has better tool support
(especially in IntelliJ) than the groovy DSL.
Using Kotlin should make refactorings easier
and the build files (hopefully) easier to understand.

Already tested:
- STF (docker)
- Build/Test
- IntelliJ Project Import
- Test STF (via Eclipse)
- Eclipse project Import

### CODE REVIEW

**Comment by stefaus** [comment]
As mentioned in Stand-Up, prepareEclipse fails:
> Task with path ':saros.picocontainer:cleanEclipseProject' not found in root project 'saros'.

**Comment by m273d15** [comment]
@stefaus Thank you, I forgot to add `apply plugin: 'eclipse'`. I will test the eclipse setup + stf via eclipse.

**Comment by m273d15** [comment]
Added the eclipse plugin, tested the eclipse import, and running STF via eclipse.

**Comment by stefaus** [comment]
I noticed that `activation-1.1.jar, mail-1.4.jar, xmlpull-1.1.3.1.jar, xpp3_min-1.1.4c.jar` are now part of the `saros.core.jar/lib` folder and `gradlew build` now fails with PMD rule violations in the subtask `:saros.picocontainer:pmdMain`.

**Comment by m273d15** [comment]
>  noticed that activation-1.1.jar, mail-1.4.jar, xmlpull-1.1.3.1.jar, xpp3_min-1.1.4c.jar are now part of the saros.core.jar/lib folder

Thank you, I had to move the `isTransitive = false` definition of the `releaseDep` config in the corresponding definition closure (I am not 100% sure why).

> gradlew build now fails with PMD rule violations in the subtask :saros.picocontainer:pmdMain.

With the new config, the `pmd` plugin is also configured in the picocontainer project. I will remove the plugin config.

**Comment by m273d15** [comment]
Rebased the commits and added the required changes

**Comment by m273d15** [comment]
Are there further change requests? Otherwise, I would rebase and merge.

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+    }
+
+    register("sarosEclipse", Copy::class) {
+        dependsOn(
+                ":saros.picocontainer:test",
+                ":saros.core:test",
+                ":saros.eclipse:test",
+                ":saros.eclipse:jar")
+
+        group = "Build"
```
[the remark this answers is not in the document]
Comment (tobous) [reply build.gradle.kts:244]: Not necessary. I was just curious. "Delivery" made sense to me as a group name as this also creates the packaged artifacts that can be used to deploy Saros.

**Pull request merged**